doc/plugins/authoring.rst

```diff
@@ -0,0 +1,211 @@
+====
+
+NixOps plugins extend NixOps core to support additional hosting
+providers and resource types.
+
+Some example plugins include:
+
+- https://github.com/NixOS/nixops-aws
+- https://github.com/NixOS/nixops-hetzner
+- https://github.com/nix-community/nixops-vbox
+- https://github.com/nix-community/nixops-libvirtd
+- https://github.com/nix-community/nixops-datadog
+
+This guide is light on the details, and intends to describe just the
+supported hooks and integration process.
+
+Packaging with Poetry and poetry2nix
+====
+
+NixOps and its plugins are packaged as standard Python applications.
+Most packages will use `Poetry <https://python-poetry.org>`_ and
+`poetry2nix <https://github.com/nix-community/poetry2nix>`_ for
+packaging with Nix.
+
+Note: NixOps is formatted with ``black`` and strictly typechecked with
+``mypy``. Your project should follow these guidelines as well, and use
+a mypy configuration at least as strict as the NixOps mypy
+configuration.
+
+First, create a ``pyproject.toml`` (see `PEP-0517
+<https://www.python.org/dev/peps/pep-0517/>`_ to describe your
+project. This is intsead of a ``setup.py``, and using both may cause
+confusing build errors. Only use a ``pyproject.toml``::
+
+  [tool.poetry]
+  name = "nixops_neatcloud"
+  version = "1.0"
+  description = "NixOps plugin for NeatCloud"
+  authors = ["Your Name <your.name@example.com>"]
+  license = "MIT"
+  include = [ "nixops_neatcloud/nix/*.nix" ]
+
+  [tool.poetry.dependencies]
+  python = "^3.7"
+  nixops = {git = "https://github.com/NixOS/nixops.git", rev = "master"}
+
+  [tool.poetry.plugins."nixops"]
+  neatcloud = "nixops_neatcloud.plugin"
+
+  [build-system]
+  requires = ["poetry>=0.12"]
+  build-backend = "poetry.masonry.api"
+
+Now create your first ``poetry.lock`` file with ``poetry lock``::
+
+  nixops_neatcloud$ nix-shell -p poetry
+  [nix-shell:nixops_neatcloud]$ poetry lock
+  Creating virtualenv nixops_neatcloud-FrXThxiS-py3.7 in ~/.cache/pypoetry/virtualenvs
+  Updating dependencies
+  Resolving dependencies... (2.1s)
+
+  Writing lock file
+
+Exit the Nix shell, and create the supporting Nix files.
+
+Create a ``default.nix``::
+
+  { pkgs ? import <nixpkgs> {} }:
+  let
+    overrides = import ./overrides.nix { inherit pkgs; };
+  in pkgs.poetry2nix.mkPoetryApplication {
+    projectDir = ./.;
+    overrides = pkgs.poetry2nix.overrides.withoutDefaults overrides;
+  }
+
+And a minimal ``overrides.nix``::
+
+  { pkgs }:
+
+  self: super: {
+    nixops = super.nixops.overridePythonAttrs({ nativeBuildInputs ? [], ... }: {
+      format = "pyproject";
+      nativeBuildInputs = nativeBuildInputs ++ [ self.poetry ];
+    });
+  }
+
+and finally, a ``shell.nix``::
+
+  { pkgs ? import <nixpkgs> {} }:
+  let
+    overrides = import ./overrides.nix { inherit pkgs; };
+  in pkgs.mkShell {
+    buildInputs = [
+      (pkgs.poetry2nix.mkPoetryEnv {
+        projectDir = ./.;
+        overrides = pkgs.poetry2nix.overrides.withoutDefaults overrides;
```

Comment:
Why `withoutDefaults` ?

Reply:
I'm not sure, I copy-pasted it from the wrong section of the poetry2nix docs by misake :) Can you fix it?